### PR TITLE
added satisfy, char, whiteSpace, skipSpaces, oneOf, noneOf, and Eq instance

### PR DESCRIPTION
**Pull request by Thimoteus**
also, I bumped up the dependency versions.

### CODE REVIEW

File: src/Text/Parsing/StringParser/String.purs [hunk 1]
```diff
@@ -46,3 +56,35 @@ string :: String -> Parser String
...
+satisfy :: (Char -> Boolean) -> Parser Char
+satisfy f = try do
+  c <- anyChar
+  if f c
+     then return c
+     else fail "Character did not satisfy predicate"
+
+-- | Match the specified character.
+char :: Char -> Parser Char
+char c = satisfy (== c)
```
Comment (paf31) [inline src/Text/Parsing/StringParser/String.purs]: Perhaps add a better error message here?

File: src/Text/Parsing/StringParser/String.purs [hunk 2]
```diff
@@ -46,3 +56,35 @@ string :: String -> Parser String
...
+
+-- | Match many whitespace characters.
+whiteSpace :: Parser String
+whiteSpace = do
+  cs <- many $ satisfy \ c -> c == 'n' || c == 'r' || c == ' ' || c == '\t'
+  return $ foldMap toString cs
```
Comment (paf31) [inline src/Text/Parsing/StringParser/String.purs]: Same here please.

File: src/Text/Parsing/StringParser/String.purs [hunk 3]
```diff
@@ -46,3 +56,35 @@ string :: String -> Parser String
...
+
+-- | Skip many whitespace characters.
+skipSpaces :: Parser Unit
+skipSpaces = do
+  whiteSpace
```
Comment (paf31) [inline src/Text/Parsing/StringParser/String.purs]: Use `void`?

**Comment by paf31** [comment]
Looks good generally, thanks, just see my couple of comments please.

File: src/Text/Parsing/StringParser/String.purs (hunk 1 above)
Comment (Thimoteus) [inline src/Text/Parsing/StringParser/String.purs]: How about this:
```purescript
fail $ "Character " <> toString c <> " did not satisfy predicate"
```
I'm not sure what other information is possible to give other than the character.

File: src/Text/Parsing/StringParser/String.purs (hunk 2 above)
Comment (Thimoteus) [inline src/Text/Parsing/StringParser/String.purs]: Since it calls satisfy, were you thinking of a different error message?

Comment (paf31) [inline src/Text/Parsing/StringParser/String.purs]: Yes, I think your `satisfy` error is fine, but maybe you could use `<?>` to provide better errors here.

**Comment by Thimoteus** [comment]
updated!

**Comment by paf31** [comment]
:+1: Thanks! Curious, why did you want the `Eq` instance?

**Comment by Thimoteus** [comment]
It was issue #4 , I figured it wouldn't hurt.

**Pull request merged**